core/src/main/java/io/grpc/internal/AbstractManagedChannelImplBuilder.java

```diff
@@ -535,22 +537,52 @@ public ManagedChannel build() {
     temporarilyDisableRetry = false;
     if (statsEnabled) {
       temporarilyDisableRetry = true;
-      CensusStatsModule censusStats = this.censusStatsOverride;
-      if (censusStats == null) {
-        censusStats = new CensusStatsModule(
-            GrpcUtil.STOPWATCH_SUPPLIER, true, recordStartedRpcs, recordFinishedRpcs,
-            recordRealTimeMetrics);
+      ClientInterceptor statsInterceptor = this.censusStatsInterceptor;
+      if (statsInterceptor == null) {
+        try {
+          Class<?> censusStatsAccessor = Class.forName("io.grpc.census.CensusStatsAccessor");
+          Method getClientInterceptroMethod =
+              censusStatsAccessor.getDeclaredMethod(
+                  "getClientInterceptor",
+                  Supplier.class,
+                  boolean.class,
+                  boolean.class,
+                  boolean.class, boolean.class);
+          statsInterceptor =
+              (ClientInterceptor) getClientInterceptroMethod
+                  .invoke(
+                      null,
+                      GrpcUtil.STOPWATCH_SUPPLIER,
+                      true,
+                      recordStartedRpcs,
```

Comment:
Doesn't have to be now, but we'll probably want to re-think how this configuration in tests is done. Could be as simple as having a method on the builder that disables the census interceptors and then requiring they be added back manually by the test code.

Reply:
Previously this is done with `AbstractManagedChannelImplBuilder#overrideCensusStatsModule(CensusStatsModule)` by injecting a `CensusStatsModule` with custom implementation for testing. Now, I changed it to `AbstractManagedChannelImplBuilder#setCensusStatsInterceptor(ClientInterceptor)`. Same idea, directly use the injected interceptor to replace the default census interceptor.